pkg/cluster/resources.go

```diff
@@ -88,6 +88,10 @@ func CreateSecret(ctx context.Context, cli client.Client, name, namespace string
 // If the configmap already exists, it will be updated with the merged Data and MetaOptions, if any.
 // ConfigMap.ObjectMeta.Name and ConfigMap.ObjectMeta.Namespace are both required, it returns an error otherwise.
 func CreateOrUpdateConfigMap(ctx context.Context, c client.Client, desiredCfgMap *corev1.ConfigMap, metaOptions ...MetaOptions) error {
+	if applyErr := ApplyMetaOptions(desiredCfgMap, metaOptions...); applyErr != nil {
```

Comment:
should this be called after "`if desiredCfgMap.GetName() == "" || desiredCfgMap.GetNamespace() == "" {`" ?

Reply:
I think if we change the order you would never be able to pass MetaOptions which could set those fields (i.e. InNamespace)

Reply:
right, `GetNamespace` will be different with the order or calls.